docs/Chef-CLI-Catalog.md

```diff
@@ -0,0 +1,228 @@
+    exec                    Runs the command in context of the embedded ruby
+    shell-init              Initialize your shell to use Chef Workstation as your primary Ruby
+    generate                Generate a new repository, cookbook, or other component
+    describe-cookbook       Prints cookbook checksum information used for cookbook identifier
+    policy                  Manage Chef Policyfiles
+    hab                     Commands relating to Chef Habitat
+    inspec                  Tests and auditing your applications and infrastructure
+    kitchen                 Test cookbooks across any combination of platforms and test suites
+    knife                   An interface between a local chef-repo and the Chef Infra Server
+    ohai                    Collect system configuration data
+    client                  Agent that runs locally on every node that is under management by Chef Infra Server
+    run                     Execute ad-hoc tasks using Chef
+    apply                   Executable program that runs a single recipe
+    shell                   A recipe debugging tool that allows the use of breakpoints within recipes
+    berks                   Dependency manager for Chef cookbooks
+    cookstyle               Code linting tool that helps you write better Chef Infra cookbooks
+
+Relocated Commands:
+    install                 Use now: chef policy install
+    update                  Use now: chef policy update
+    push                    Use now: chef policy push
+    push-archive            Use now: chef policy push-archive
+    show-policy             Use now: chef policy show
+    diff                    Use now: chef policy diff
+    export                  Use now: chef policy export
+    clean-policy-revisions  Use now: chef policy clean revisions
+    clean-policy-cookbooks  Use now: chef policy clean cookbooks
+    delete-policy-group     Use now: chef policy group delete
+    delete-policy           Use now: chef policy delete
+    undelete                Use now: chef policy undelete
+
+Experimental Commands:
+    analyze                 Analyze artifacts from a Chef Infra Server
+```
+
+### EXTRA: Present correct usage of sub-binaries
+
+An additional task from this proposal is the use of an environment variable that
+can control the "Usage" message of sub-binaries. Currently, the `chef-analyze`
+binary displays the usage message `chef-analyze [command]` when it is run directly
+and also when it is run from the `chef` catalog. We would like the usage message
+to be consistent with the way users execute the commands so if you run `chef analyze`
+the usage sould be `chef analyze [command]`; and if you run `chef-analyze` the
+usage should continue to be `chef-analyze [command]`.
+
+## Open points for further discussion:
+* Are we discouraging the use of individual sub-binaries like `knife` or `chef-run`?
+* What happens when a user runs a sub-binary without the prefix `chef`?
```

Comment:
IMO, we should not communicate the existence of these binaries to users but we should not prevent their usage. It is just unsupported usage.

Reply:
I misunderstood this line - I was thinking that we should document `chef policy` and not document `chef-policy` but if they run `chef-policy` we don't do anything to prevent them from doing that.